File: domain/src/main/java/com/sfr/domain/repository/NumberRepository.java
package com.sfr.domain.repository;

import com.sfr.domain.model.NumberInformationModel;
import com.sfr.domain.model.NumberModel;
import com.sfr.domain.model.UserNumberHistory;
import com.sfr.domain.result.Result;

public interface NumberRepository {
    void getNumberInformation(NumberModel numberModel, Result<NumberInformationModel> result);
    void getRandomNumberInformation(Result<NumberInformationModel> result);
    void getUserNumbersHistory(Result<NumberInformationModel> result);
    void saveUserNumberHistory(UserNumberHistory userNumberHistory, Result<UserNumberHistory> result);
}

File: domain/src/main/java/com/sfr/domain/repository/NumberRepository.kt
package com.sfr.domain.repository

import com.sfr.domain.model.NumberInformationModel
import com.sfr.domain.model.NumberModel
import com.sfr.domain.model.UserNumberHistory

interface NumberRepository {

    fun getNumberInformation(numberModel: NumberModel, result: (NumberInformationModel) -> Unit, wasException: (Exception) -> Unit)
    fun getRandomNumberInformation(result: (NumberInformationModel) -> Unit, wasException: (Exception) -> Unit)
    fun getUserNumbersHistory(result: (List<UserNumberHistory>) -> Unit, wasException: (Exception) -> Unit)

    fun saveUserNumberHistory(userNumberHistory: UserNumberHistory, result: (List<UserNumberHistory>) -> Unit, wasException: (Exception) -> Unit)

}